tox.ini

```diff
@@ -68,8 +70,8 @@ commands_pre =
     {[testenv:integrationtests]commands_pre}
 commands =
     flake8
-    {[testenv:coverage]commands}
-    {[testenv:unittests]commands}
+    # {[testenv:coverage]commands}
+    # {[testenv:unittests]commands}
```

Comment:
Can you please explain why you commented these out?

Reply:
On [this PR](https://github.com/gridsingularity/gsy-myco-sdk/pull/55), Hassan removed the unit tests from this repository. The checks were failing as there was nothing to run the coverage checks against to, as you can see [here](https://github.com/gridsingularity/gsy-myco-sdk/runs/6376944286?check_suite_focus=true).

Reply:
@Hassan754 out of curiosity: why do we have two folders `tests/` and `unit_tests/` in the myco repo?

Also, do we have plans to implement unit tests for the myco?

Reply:
> On [this PR](https://github.com/gridsingularity/gsy-myco-sdk/pull/55), Hassan removed the unit tests from this repository. The checks were failing as there was nothing to run the coverage checks against to, as you can see [here](https://github.com/gridsingularity/gsy-myco-sdk/runs/6376944286?check_suite_focus=true).

@ERafaelMartinez shouldn't tests have failed in the CI of that PR from Hassan? Also, it's a bit strange. `tox -e coverage` fails, but if I give the separate commands from the `coverage` environment everything seems to work smoothly.

Reply:
> shouldn't tests have failed in the CI of that PR from Hassan?

@fievelk That is actually the case.

> Also, it's a bit strange. tox -e coverage fails, but if I give the separate commands from the coverage environment everything seems to work smoothly.

For me, the first command of the environment `coverage run -m pytest unit_tests/ --random-order {posargs:tests}` produces the same error. The other 3 run ok. 
I could change this to comment out just the command producing the error...

Reply:
Some questions:
- did you run the exact command you pasted on your console? If so, I guess it should fail because `{posargs:tests:}` has no meaning out of a `tox` command (so the `pytest` command will try to use it literally)
- what is `{posargs:tests}` supposed to do here? Aren't we specifying `unit_tests/` as a folder already?

If I run `coverage run -m pytest unit_tests/ --random-order` (without `tox`) it works smoothly. With tox, it fails for some reason.

Reply:
> did you run the exact command you pasted on your console? If so, I guess it should fail because {posargs:tests:} has no meaning out of a tox command (so the pytest command will try to use it literally)

Ah yes, you are right. If I take that out I see no error, yet the CoverageWarning persists. I think this is the reason why tox fails. According to the [docs](https://tox.wiki/en/latest/config.html#:~:text=Commands%20will%20execute%20one%20by%20one%20in%20sequential%20fashion%20until%20one%20of%20them%20fails%20(their%20exit%20code%20is%20non%2Dzero)%20or%20all%20of%20them%20succeed.), it will fail with an exit code other than 0. The exit code after the warning is 5, according to the logs. 

> what is {posargs:tests} supposed to do here? Aren't we specifying unit_tests/ as a folder already?

Not sure actually, I guess it adds a second folder to look for tests into. Yet, if you try `coverage run -m pytest unit_tests/ --random-order tests` on the CLI, outcome is the same.

Reply:
I see, we do not have any tests left that we can run `coverage` on, which makes it fail.
The it is ok for me to comment them out (with a small sentence to explain why).
We definitely should add unit tests (ideally in the `tests` folder), at least a few, in oder to have a working workflow. But this needs to be tackled in a separate story. Feel free to add one @ERafaelMartinez

Reply:
@Hassan754 I have a question for you since we're talking about this. I saw you created a`unit_tests/` folder in the myco repository (also with the `__init__.py` module). Why was this necessary? Could we use the `tests/` folder instead?

Reply:
@hannesdiedrich [TESTS-152](https://gridsingularity.atlassian.net/browse/TESTS-152) :)